src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
+enum ShapeType {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 export interface Figure {
+  shape: ShapeType,
+  color: 'red' | 'green' | 'blue',
+  area: number,
+}
 
+function calculateArea(a: number, b: number, c: number): number {
+  const s: number = 1 / 2 * (a + b + c);
+
+  return Math.sqrt(s * (s - a) * (s - b) * (s - c));
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public readonly shape: ShapeType = ShapeType.Triangle;
+
+  public readonly area: number;
 
+  constructor(
+    public color: Figure['color'],
+    a = 0,
+    b = 0,
+    c = 0,
+  ) {
+    this.color = color;
+    this.area = calculateArea(a, b, c);
+
+    if (a <= 0 || b <= 0 || c <= 0
+      || a >= b + c
+      || b >= a + c
+      || c >= b + a) {
+      throw new Error('');
```

Comment:
explain the error

Reply:
also use conditions before setting values

you don't need to initialize smth if condition is wrong